File: frontend/src/types/auditLog.ts
export type AuditLogStatus = 'SUCCESS' | 'FAILED';

export interface AuditLogQueryParams {
  tenantId?: number;
  actorUsername?: string;
  action?: string;
  entityType?: string;
  status?: AuditLogStatus | '';
  startTime?: string;
  endTime?: string;
  limit?: number;
}

export interface AuditLogEntry {
  id: number;
  tenantId: number;
  actorUsername?: string;
  actorUserId?: number;
  action: string;
  entityType: string;
  entityId?: string;
  detail?: string;
  status: AuditLogStatus;
  requestId?: string;
  clientIp?: string;
  userAgent?: string;
  durationMs?: number;
  method?: string;
  path?: string;
  httpStatus?: number;
  errorCode?: string;
  errorMessage?: string;
  authTenantId?: number;
  authTenantCode?: string;
  crossTenant: boolean;
  createdAt: string;
}

File: backend/xuan-audit/src/main/java/com/xuan/erp/audit/domain/model/AuditLog.java
package com.xuan.erp.audit.domain.model;

import com.xuan.erp.audit.domain.model.type.AuditWriteStatus;
import java.time.OffsetDateTime;

public record AuditLog(
        Long id,
        Long tenantId,
        String actorUsername,
        Long actorUserId,
        String action,
        String entityType,
        String entityId,
        String detail,
        AuditWriteStatus status,
        String requestId,
        String clientIp,
        String userAgent,
        Long durationMs,
        String method,
        String path,
        Integer httpStatus,
        String errorCode,
        String errorMessage,
        Long authTenantId,
        String authTenantCode,
        boolean crossTenant,
        OffsetDateTime createdAt
) {

    public AuditLog withId(Long id) {
        return new AuditLog(
                id,
                tenantId,
                actorUsername,
                actorUserId,
                action,
                entityType,
                entityId,
                detail,
                status,
                requestId,
                clientIp,
                userAgent,
                durationMs,
                method,
                path,
                httpStatus,
                errorCode,
                errorMessage,
                authTenantId,
                authTenantCode,
                crossTenant,
                createdAt
        );
    }
}
